db/db_impl.cc

```diff
@@ -2296,15 +2296,17 @@ Status DBImpl::FlushMemTable(ColumnFamilyData* cfd,
       return Status::OK();
     }
 
-    WriteThread::Writer w;
-    write_thread_.EnterUnbatched(&w, &mutex_);
+    if (!cfd->mem()->IsEmpty()) {
+      WriteThread::Writer w;
+      write_thread_.EnterUnbatched(&w, &mutex_);
 
-    // SwitchMemtable() will release and reacquire mutex
-    // during execution
-    s = SwitchMemtable(cfd, &context);
-    write_thread_.ExitUnbatched(&w);
+      // SwitchMemtable() will release and reacquire mutex
+      // during execution
+      s = SwitchMemtable(cfd, &context);
+      write_thread_.ExitUnbatched(&w);
 
-    cfd->imm()->FlushRequested();
+      cfd->imm()->FlushRequested();
```

Comment:
I thought FlushRequested() is bound with SwitchMemtable(), that is, memtable switched, flush requested. And in all other places, where SwitchMemtable() invoked, FlushRequested() invoked. So no flush request will miss.
Of cause, put this line out of the if statement would be OK.